### PR TITLE
[PublishTestResultsV2] Add option to fail task if no test result files are found

### PR DESCRIPTION
**Pull request by walliski**
**Task name**: PublishTestResultsV2

**Description**:
Adds an option that lets the user configure the task to fail in case no
test result files have been found. This serves as a safeguard to make
sure that the test-run task and this task has been configured properly
to find the test result files.

We are defaulting to not failing the task on missing files, to keep current behavior.

**Documentation changes required:** Y
I guess this option should be documented.

**Added unit tests:** N
Not sure how this should be tested, as the "Warning" functionality doesn't seem to be tested either?

**Attached related issue:** Y
Solves microsoft/azure-pipelines-tasks#13275

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

I have no idea if this works, or how to test it. I have ran the build and existing tests for the task, but haven't manually tested the task anywhere (and have no idea how anything here works...).

### CODE REVIEW

File: Tasks/PublishTestResultsV2/publishtestresults.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ function isNullOrWhitespace(input: any) {
     return input.replace(/\s/g, '').length < 1;
 }
 
-function publish(testRunner, resultFiles, mergeResults, failTaskOnFailedTests, platform, config, runTitle, publishRunAttachments, testRunSystem) {
+function publish(testRunner, resultFiles, mergeResults, failTaskOnFailedTests, failTaskOnMissingResultsFile, platform, config, runTitle, publishRunAttachments, testRunSystem) {
```
Comment (walliski) [inline Tasks/PublishTestResultsV2/publishtestresults.ts:19]: Not sure if this is needed?

**Comment by walliski** [comment]
Soooo is this going somewhere?

**Comment by Alexander-Bartosh** [comment]
Why is that a problem to review this nice improvement for 7 months !?

**Comment by mathiash98** [comment]
@shailesh-sk could you do a review on this PR?

**Comment by mathiash98** [comment]
@shailesh-sk Could you do a review on this PR.

Or maybe some of the more active contributers could do a review?
Perhaps @anatolybolshakov or @nadesu @v-jkarri


Merging this would allow closing https://github.com/microsoft/azure-pipelines-tasks/issues/13275 and https://github.com/microsoft/azure-pipelines-tasks/issues/13275

**Comment by sonnyhcl** [comment]
reping here since it's a really really useful feature! could anyone go through the code review?

**Comment by the-rule** [comment]
Could we merge this essential feature any time soon?

**Comment by walliski** [comment]
@the-rule I can't really do anything about it, its up for the maintainers to merge it.

To me its infuriating that such a simple change, a feature that exists in pretty much all CI systems, has not been merged yet, and has been actively ignored for over a year.

**Comment by the-rule** [comment]
@shailesh-sk ping-ping

**Comment by the-rule** [comment]
> To me its infuriating that such a simple change, a feature that exists in pretty much all CI systems, has not been merged yet, and has been actively ignored for over a year.

Totally agree, to me this change is not even a feature but a bug fix!

**Comment by Alexander-Bartosh** [comment]
@shailesh-sk I would completely agree with @the-rule !

**Comment by JoostvdB94** [comment]
Maybe anyone else who can review this PR? @DenisRumyantsev?
Very strange that such a small improvement is stuck on review.

**Comment by eholman** [comment]
@navin22 can you maybe help? This is such a small PR with incredibly added value...

**Comment by eholman** [comment]
@navin22 or @rasunkar is there no one to be found?

**Comment by walliski** [comment]
Beep boop

**Comment by triptijain2112** [comment]
Merged in Pull request: https://github.com/microsoft/azure-pipelines-tasks/pull/18931
This would be available in version 2.228.0

**Pull request closed without merging**